simple-acquaintances/data/datagen.py

```diff
@@ -0,0 +1,395 @@
+        """
+        if self.outputDir:
+            if self.outputDir[0] == '~':
+                prefix = os.path.expanduser(self.outputDir)
+            else:
+                prefix = self.outputDir
+        else:
+            prefix = ''
+
+        for key in data:
+            # We have the same keys for data and its corresponding path.
+            with open(os.path.join(prefix, paths[key]), 'w') as f:
+                if key == 'config':
+                    json.dump(data[key], f, indent = 4, sort_keys = True)
+                else:
+                    for dataItem in data[key]:
+                        f.write("\t".join([str(d) for d in dataItem]) + "\n")
+
+    def __prepareDataToWrite(self, knowsMatrix, people):
+        """
+        Prepare data for writing to file.
+        """
+        n = knowsMatrix.shape[1]
+
+        knowsData = []
+        knowsObs = []
+        knowsTarget = []
+        knowsTruth = []
+        
+        totalCombinations = knowsMatrix.shape[0] * (knowsMatrix.shape[1] - 1)
+        targets = np.random.choice(range(totalCombinations), 
+                                    int(self.targetSplit * totalCombinations), replace = False)
+        for i in range(knowsMatrix.shape[0]):
+            for j in range(knowsMatrix.shape[1]):
+                if i != j:
+                    knowsData.append([i, j, knowsMatrix[i][j]])
+                if (n * i + j) not in targets and i != j:
+                    knowsObs.append([i, j, knowsMatrix[i][j]])
+                elif (n * i + j) in targets and i != j:
+                    knowsTarget.append([i, j])
+                if (n * i + j) in targets and i != j:
+                    knowsTruth.append([i, j, knowsMatrix[i][j]])
+
+        livedObs = []
+        for person in people:
+            for index, place in enumerate(person.lived):
+                if place == 1:
+                    livedObs.append([person.index, index])
+
+        likesObs = []
+        for person in people:
+            for globalLike in person.globalLikes:
+                likesObs.append([person.index, globalLike, person.globalLikes[globalLike]])
+
+        # NOTE: Both global and local likes are combined in a single list called likesObs.
+        for person in people:
+            for localLike in person.localLikes:
+                likesObs.append([person.index, len(self.globalThings) + localLike, person.localLikes[localLike]])
+        
+        config = {  'People': self.numberOfPeople,
+                    'Places': len(self.places),
+                    'Global things': len(self.globalThings),
+                    'Local things': len(self.localThings),
+                    'Places lived mean': self.placesLivedMean,
+                    'Places lived standard deviation': self.placesLivedSD,
+                    'Local likes variance': self.localLikesVariance,
+                    'Target split': self.targetSplit,
+                    'Coin flip seed': self.coinFlipSeed,
```

Comment:
This defaults to `None`, but that would screw us over if someone wanted to reproduce the results of the generation and they didn't specify the seed.
Remember, the point of the config/options file is so a person can look at the data directory and know how it was generated.

So, what you should do is generate a seed if one is not specified.
Do this using an unseeded RNG (it is actually seeded, just not by you).
Ex: `np.random.randint(2**32)`.

Reply:
Aah I see. Let's talk about this in person before I make any further changes.